cw-orch-daemon/src/builder.rs

```diff
@@ -87,32 +92,70 @@ impl DaemonAsyncBuilder {
         self
     }
 
+    /// Specifies path to the daemon state file
+    /// Defaults to env variable.
+    ///
+    /// Variable: STATE_FILE_ENV_NAME.
+    ///
+    /// This field is ignored for rebuilt daemon and path of the original daemon used instead
+    pub fn state_path(&mut self, path: impl ToString) -> &mut Self {
```

Comment:
For multi-daemon cases it's nice to have this method for flexibility IMO. It gives more straightforward configuration than setting/unsetting env var. 
Where I see it's useful:
- You don't want your daemons fight over file. For example, use separate file for each chain
- Multiple scripts for same contract. For example one script is mainnet and one is testnet and you don't want to use same file for test-chains and main-chains, it also should be fine to run them simultaneously

Reply:
You're suggesting with this solution for the Daemon to use multple state file for multiple chains, this beats the purpose of the state file which is to have eveyrthing in one place. Its structure is perfect in that way. Especially for multi-chain apps, this is really a must-have.
I don't think it's necessary, because we developped this lock especially to work with the same file !

Reply:
Only issue is two different binaries won't be able to access same file because of lock. Anyway it's out of scope of this PR so made it private and unused for now